### ISSUE TITLE
[Not a bug] Thank you!

### ISSUE THREAD

**Issue by bketelsen** [RicardoEPRodrigues/magicmouse-hid#1]
Just wanted to drop by and say "thanks!" this fixed my problems and nothing else worked.  You rock for sharing it.

**Comment by adw0rd** [comment]
@RicardoEPRodrigues Thank you!

**Comment by RicardoEPRodrigues** [comment]
I only now saw this. It was unexpected and amazing! Thank you for taking the time to come here and post this! 

This driver is based off of the work of @robotrovsky, @svartalf, @0xABAD and probably others. Thanks to them as well. 😁😁😁

**Comment by glide-the** [comment]
i'm english too bad， Just wanted to drop by and say "thanks!" for fixYou rock for sharing it.

同样，我也想表达这个感谢， 前段时间就注意到了这个问题，原仓库 **rohitpid/Linux-Magic-Trackpad-2-Driver**中覆盖了仓库**mwyborski/Linux-Magic-Trackpad-2-Driver**提交至 4.20内核中的已有的模块，在发现重启后无法加载并且dmesg发现是内核有签名验证，为此我看了不少文档，但是没有找到方法，最终我放弃了，
过去三个月后发现角落里的鼠标。。。然后想了想，要不在研究研究hid怎么加载的，偶然间，我发现了这个仓库。
发现还是补丁香，哈哈哈

**Comment by mvdve** [comment]
Same! Thanks for this amazing work. I tried to get my mouse working for over two years and had given up. Checked today if there was something new by now, found this project, installed the deb package and BAM! works. So happy!

**Comment by Ravisutha** [comment]
Many thanks, @RicardoEPRodrigues. It worked like a charm!

**Comment by briantopping** [comment]
Indeed, this is critical work. Thank you for bridging "the rest of us" toward 5.15!! I hope the best of everything is coming to you!

**Comment by greg-fischer** [comment]
THANK YOU!! 

I have been trying for days to get the mouse working and STOP SCROLLING when I move the mouse!  I almost gave up.  Why is there no mention of this driver out there?  All I keep finding is the same scroll sensitivity, which is NOT the same thing. Its the touch causing a scroll too easily.  I was going nuts trying to get any work done!  Wow! Thank you!   I going to post this on my web.

I'm trying to move away from MacOS to Linux for my main work system.  I can do 95%+ just fine on Linux now, to my surprise. (and I had done this for a year about 10 years back, and moved to Mac)  But I just can't give up my Apple KB/trackpad/mouse combo.  Its just too drilled into my productivity.  And it was the biggest fear I had switching, that it wouldn't work.  

Its still there, the touch/scroll thing, but about 10x less!  And I see the module params for scroll_delay...    I'll tweak this and see.  

~  If I can just get KDE to allow the second virtual screen to stay put when swiping the primary screen...  MacOS almost has no advantages for me.

Thanks again!!
